docs/contributing.rst

```diff
@@ -179,6 +179,25 @@ Scrapy:
   See https://help.github.com/en/github/using-git/setting-your-username-in-git for
   setup instructions.
 
+.. _pre-commit:
+
+Pre-commit
+==========
+
+Pre-commit is a tool that allows developers to specify a set of checks to be run 
+automatically every time they make a commit. This can include code style checks, 
+linting, and automated tests. 
+The checks are defined in a configuration file called .pre-commit-config.yaml.
+
+By using pre-commit, developers can ensure that their code adheres to a consistent 
+style and passes certain tests before they make a commit. 
+This can help catch errors early in the development process and prevent them 
+from being pushed to the main branch.
+
+To use pre-commit, developers first need to install it on their local machine. 
+Once it is installed, they can specify the checks they want to run in the .pre-commit-config.yaml file. 
+After that, pre-commit will run automatically every time they make a commit.
```

Comment:
I don‘t think we need to focus so much on how to configure pre-commit for the repository, i.e. `.pre-commit-config.yaml` is already in the repository, and instead focus only on what contributors need to do to get it working. For example:

```suggestion
We use `pre-commit`_ to automatically address simple code issues before every 
commit.

Before you start writing a patch:

#.  `Install pre-commit <https://pre-commit.com/#installation>`_.

#.  On the root of your local clone of the Scrapy repository, run the following 
    command:

    .. code-block:: bash

       pre-commit install

Now our pre-commit hooks will run every time you create a Git commit. Upon 
finding issues, pre-commit hooks aborts your commit, and they either fix 
the corresponding issues automatically or only report them to you. If they fix 
the issues automatically, creating your commit again should succeed. Otherwise, 
you may need to address the corresponding issues manually first.
```

Also, I think it may be good to merge the first 2 bullet points from the section above, about code style, into a single one that indicates that we use black for code formatting, and that it can be executed manually with `tox` (and indicate how) or automatically with pre-commit.

Reply:
Mind the spacing when you apply the fix, empty lines make a different in reStructuredText.

Reply:
```
/home/runner/work/scrapy/scrapy/docs/contributing.rst:193:Unexpected indentation.
```

Reply:
Today I Learned that copying code from GitHub suggestions breaks spacing. Sorry about that.

This should be easier to copy:

```
We use `pre-commit`_ to automatically address simple code issues before every 
commit.

.. _pre-commit: https://pre-commit.com/

Before you start writing a patch:

#.  `Install pre-commit <https://pre-commit.com/#installation>`_.

#.  On the root of your local clone of the Scrapy repository, run the following 
    command:

    .. code-block:: bash

       pre-commit install

Now pre-commit will check your changes every time you create a Git commit. Upon 
finding issues, pre-commit aborts your commit, and either fixes those issues 
automatically, or only reports them to you. If it fixes those issues 
automatically, creating your commit again should succeed. Otherwise, you may 
need to address the corresponding issues manually first.
```

The rendering is significantly different than what you have at the moment (e.g. “Before you start…” is interpreted as a definition list term in your current version).

I also included the link target for the “pre-commit” link at the beginning, that I missed in my suggestion, which I assume is why you ended up adding a reference to the header, and that is where the text now links. Mind that you need to remove the header reference line (the ```.. _pre-commit:`` line before the header) I would keep the reference that you added to the header, but this link should point to the pre-commit website instead.